operator-common/src/main/java/io/strimzi/operator/common/metrics/KindNamespaceMetricKey.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.common.metrics;
+
+/**
+ * Metric key for metrics specified by kind and namespace.
+ */
+public class KindNamespaceMetricKey extends MetricKey {
```

Comment:
Not necessary, if you prefer, ill delete it and make the abstract non-abstract

Reply:
I don't have a problem with the abstract class as much as with how it is implemented. The way the abstract class looks today is strange because it has the kind and namespace already in it yet it needs the non-abstract class to override the key. I would do one of these:
* Keep the abstract class, but move the kind and namespace out of it
* Move the `getKey` implementation to it based on the kind and namespace make it non-abstract

I do not have a strong preference. But the current state seems to be stuck somewhere in between.

Reply:
Then i'll personally prefer second option, will do that.